static/js/stream_popover.js

```diff
@@ -404,19 +407,26 @@ exports.register_stream_handlers = function () {
         e.stopPropagation();
     });
 
-    // Unstar all messages
-    $("body").on("click", "#unstar_all_messages", (e) => {
+    // Unstar all messages, or all messages in a topic
+    $("body").on("click", "#unstar_all_messages, .sidebar-popover-unstar-all-in-topic", (e) => {
         exports.hide_starred_messages_popover();
         e.preventDefault();
         e.stopPropagation();
         $(".left-sidebar-modal-holder").empty();
-        $(".left-sidebar-modal-holder").html(render_unstar_messages_modal());
+        const topic_name = $(".sidebar-popover-unstar-all-in-topic").attr("data-topic-name");
+        const args = {
+            topic_name,
+        };
+        $(".left-sidebar-modal-holder").html(render_unstar_messages_modal(args));
+        exports.hide_topic_popover();
         $("#unstar-messages-modal").modal("show");
     });
 
     $("body").on("click", "#do_unstar_messages_button", (e) => {
         $("#unstar-messages-modal").modal("hide");
-        message_flags.unstar_all_messages();
+        const topic_name = $(".sidebar-popover-unstar-all-in-topic").attr("data-topic-name");
+        const stream_id = $(".sidebar-popover-unstar-all-in-topic").attr("data-stream-id");
+        message_flags.unstar_messages(topic_name, Number(stream_id));
```

Comment:
Can you do `Number.parseInt` here? See https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/prefer-number-properties.md

eslint should have failed on this.

Reply:
> eslint should have failed on this.

I didn't understand... there's no such rule in the document you linked right?

Reply:
idk, https://chat.zulip.org/#narrow/stream/43-automated-testing/topic/unicorn.20prefer-number-properties